### PR TITLE
Beaslecd codereview2

### PR DESCRIPTION
**Pull request by beaslecd**
Analysis: This program allows the user to take a pic and create a puzzle to solve or grabs a random picture via API and creates a puzzle to solve.

- It was avaliable on time.
- The program complied and was well-documented.

Three things I've learned

- Using one MainViewModel for multiple Fragments
- You can substitivbue a 'if' statement  with a 'when' to be more clear and concise when dealing with multiple conditions in Kotlin
- Accessing widgets like buttons in the code

### CODE REVIEW

File: app/src/main/java/com/example/puzzlepic/service/PictureService.kt [hunk 1]
```diff
@@ -23,7 +23,7 @@ class PictureService {
      * Uses a callback to retrieve json data from network and handles response accordingly
      * @return an LiveArrayList of pictures.
      */
-    fun fetchPicture() : MutableLiveData<ArrayList<Picture>> {
+   internal fun fetchPicture() : MutableLiveData<ArrayList<Picture>> {
```
Comment (beaslecd) [inline app/src/main/java/com/example/puzzlepic/service/PictureService.kt]: Adding visualbility modifier 
Source: https://kotlinlang.org/docs/visibility-modifiers.html#constructors

File: app/src/main/java/com/example/puzzlepic/ui/main/MainFragment.kt [hunk 2]
```diff
@@ -42,24 +42,20 @@ class MainFragment : Fragment() {
...
     private var puzzles : ArrayList<Picture> = ArrayList<Picture>()
-
     private var _binding: MainFragmentBinding? = null
     private val binding get() = _binding!!
-
     private var user : FirebaseUser? = null
+    private lateinit var currentPhotoPath: String
+    protected var puzzleURI : Uri? = null
+    private lateinit var viewModel: MainViewModel
+    lateinit var navController: NavController
```
Comment (beaslecd) [inline app/src/main/java/com/example/puzzlepic/ui/main/MainFragment.kt]: Ordering the modifiers in the correct order.
Source: https://kotlinlang.org/docs/coding-conventions.html#modifiers-order

File: app/src/main/java/com/example/puzzlepic/ui/main/MainFragment.kt [hunk 3]
```diff
@@ -223,12 +219,12 @@ class MainFragment : Fragment() {
...
     }
 
-    fun initiateLoading() {
+    private fun initiateLoading() {
         binding.progressBar.isVisible = true
         toggleButtons("off")
     }
 
-    fun loadingComplete(){
+    private fun loadingComplete(){
```
Comment (beaslecd) [inline app/src/main/java/com/example/puzzlepic/ui/main/MainFragment.kt:227]: Adding visability modifier
Source: https://kotlinlang.org/docs/visibility-modifiers.html

File: app/src/main/java/com/example/puzzlepic/ui/main/MainFragment.kt [hunk 4]
```diff
@@ -241,70 +237,72 @@ class MainFragment : Fragment() {
...
+                    it.alpha = 0.3F
+                }
+                binding.navBarUserButton.also{
+                    it.isEnabled = false
+                    it.alpha = 0.3F
+                }
+                binding.button2.also{
+                    it.isEnabled = false
+                    it.alpha = 0.3F
+                }
```
Comment (beaslecd) [inline app/src/main/java/com/example/puzzlepic/ui/main/MainFragment.kt:261]: Replaced the if statement with the more concise, when statement.
Source: https://phauer.com/2017/idiomatic-kotlin-best-practices/

File: app/src/main/java/com/example/puzzlepic/ui/main/MainFragment.kt [hunk 5]
```diff
@@ -241,70 +237,72 @@ class MainFragment : Fragment() {
...
+                    it.alpha = 1F
+                }
+                binding.navBarUserButton.also{
+                    it.isEnabled = true
+                    it.alpha = 1F
+                }
+                binding.button2.also{
+                    it.isEnabled = true
+                    it.alpha = 1F
+                }
```
Comment (beaslecd) [inline app/src/main/java/com/example/puzzlepic/ui/main/MainFragment.kt:283]: Replaced the If statement with the more concise When statement
Source:https://phauer.com/2017/idiomatic-kotlin-best-practices/

File: app/src/main/java/com/example/puzzlepic/ui/main/MainFragment.kt [hunk 6]
```diff
@@ -241,70 +237,72 @@ class MainFragment : Fragment() {
...
+                    it.alpha = if (it.isEnabled) 1F else .3f
+                }
+                binding.navBarUserButton.also{
+                    it.isEnabled = !it.isEnabled
+                    it.alpha = if (it.isEnabled) 1F else .3f
+                }
+                binding.button2.also{
+                    it.isEnabled = !it.isEnabled
+                    it.alpha = if (it.isEnabled) 1F else .3f
+                }
```
Comment (beaslecd) [inline app/src/main/java/com/example/puzzlepic/ui/main/MainFragment.kt:305]: Replaced the If statement with the more concise When statement
Source:https://phauer.com/2017/idiomatic-kotlin-best-practices/

File: app/src/main/java/com/example/puzzlepic/ui/puzzle/PuzzleFragment.kt [hunk 7]
```diff
@@ -27,12 +27,12 @@ import java.util.*
 
 class PuzzleFragment : Fragment() {
 
+    private var _binding: PuzzleFragmentBinding? = null
+    private val binding get() = _binding!!
+    private lateinit var currentPhotoPath: String
```
Comment (beaslecd) [inline app/src/main/java/com/example/puzzlepic/ui/puzzle/PuzzleFragment.kt:34]: Reordering the modifiers and putting them in the correct order.
Source:https://kotlinlang.org/docs/coding-conventions.html#modifiers-order

File: app/src/main/java/com/example/puzzlepic/ui/puzzle/PuzzleFragment.kt [hunk 8]
```diff
@@ -129,12 +129,12 @@ class PuzzleFragment : Fragment() {
...
     }
 
-    fun initiateLoading() {
+    private fun initiateLoading() {
         binding.progressBar.isVisible = true
         toggleButtons("off")
     }
 
-    fun loadingComplete(){
+    private fun loadingComplete(){
```
Comment (beaslecd) [inline app/src/main/java/com/example/puzzlepic/ui/puzzle/PuzzleFragment.kt:492]: Adding visaiblity modifiers.
Source:https://kotlinlang.org/docs/visibility-modifiers.html

File: app/src/main/java/com/example/puzzlepic/ui/puzzle/PuzzleFragment.kt [hunk 9]
```diff
@@ -147,58 +147,60 @@ class PuzzleFragment : Fragment() {
...
+                    it.alpha = 0.3F
+                }
+                binding.navBarImageButton.also{
+                    it.isEnabled = false
+                    it.alpha = 0.3F
+                }
+                binding.navBarUserButton.also{
+                    it.isEnabled = false
+                    it.alpha = 0.3F
+                }
```
Comment (beaslecd) [inline app/src/main/java/com/example/puzzlepic/ui/puzzle/PuzzleFragment.kt:545]: Replaced the If statement with the more concise When statement
Source:https://phauer.com/2017/idiomatic-kotlin-best-practices/

File: app/src/main/java/com/example/puzzlepic/ui/puzzle/PuzzleFragment.kt [hunk 10]
```diff
@@ -147,58 +147,60 @@ class PuzzleFragment : Fragment() {
...
-                it.alpha = if (it.isEnabled) 1F else .3f
+            else -> {
+                binding.navBarCameraButton.also{
+                    it.isEnabled = !it.isEnabled
+                    it.alpha = if (it.isEnabled) 1F else .3f
+                }
+                binding.savePuzzleButton.also{
+                    it.isEnabled = !it.isEnabled
+                    it.alpha = if (it.isEnabled) 1F else .3f
+                }
```
Comment (beaslecd) [inline app/src/main/java/com/example/puzzlepic/ui/puzzle/PuzzleFragment.kt]: Replaced the If statement with the more concise When statement
Source:https://phauer.com/2017/idiomatic-kotlin-best-practices/

File: app/src/main/java/com/example/puzzlepic/ui/puzzle/PuzzleFragment.kt [hunk 11]
```diff
@@ -147,58 +147,60 @@ class PuzzleFragment : Fragment() {
...
+                binding.navBarImageButton.also{
+                    it.isEnabled = !it.isEnabled
+                    it.alpha = if (it.isEnabled) 1F else .3f
+                }
+                binding.navBarUserButton.also{
+                    it.isEnabled = !it.isEnabled
+                    it.alpha = if (it.isEnabled) 1F else .3f
+                }
```
Comment (beaslecd) [inline app/src/main/java/com/example/puzzlepic/ui/puzzle/PuzzleFragment.kt:581]: Replaced the If statement with the more concise When statement
Source:https://phauer.com/2017/idiomatic-kotlin-best-practices/

**Comment by discospiff** [comment]
Good suggestions here.  I recommend that the group review them and decide what they would like to merge.

**Review by erswagner7 — APPROVED** [review]
Accepting code review changes

**Review by erswagner7 — APPROVED** [review]

**Review by erswagner7 — APPROVED** [review]

**Review by erswagner7 — APPROVED** [review]

**Comment by erswagner7** [comment]
Accepting code review changes

**Pull request merged**